src/CppUTest/SimpleString.cpp

```diff
@@ -422,6 +422,64 @@ void SimpleString::replace(const char* to, const char* with)
         setInternalBufferAsEmptyString();
 }
 
+SimpleString SimpleString::printable() const
+{
+    const char* shortEscapeCodes[] =
+    {
+        "\\a",
+        "\\b",
+        "\\t",
+        "\\n",
+        "\\v",
+        "\\f",
+        "\\r"
+    };
+
+    size_t str_size = size();
+    size_t new_str_size = str_size;
+
+    for (size_t i = 0; i < str_size; i++)
+    {
+        unsigned char c = (unsigned char) buffer_[i];
+        if ((c >= 0x07) && (c <= 0x0D))
```

Comment:
Can you extract this if in a separate method so it is more clear what this is about... and to remove the duplication with the if below?

Reply:
I don't get your intention. Do you mean that you'd like the if conditions to be extracted as separate methods (like `if (hasShortEscapeCode(c)) ...`? Or do ypu want the if/else if/else statements to be extracted to a couple of methods like `adjustPrintableSize()`/`writePrintableChar` and in this method just keep the two loops calling these methods?

Reply:
Start with extracting the condition as they are and cryptic and duplicate. If the function is too long, we can look at extracting other parts too.

Reply:
What do you think about the new refactoring?